src/main/resources/database/postgresql/1/patch_24.sql

```diff
@@ -0,0 +1,178 @@
+-- feature/static-metrics
+-- Create a new table containing the reputation of each day grouped by day
+CREATE TABLE repbot_schema.metrics_reputation
+(
+    day   DATE   NOT NULL,
+    cause TEXT   NOT NULL,
+    count BIGINT NOT NULL
+);
+
+CREATE INDEX metrics_reputation_day_index
+    ON repbot_schema.metrics_reputation (day);
+
+ALTER TABLE repbot_schema.metrics_reputation
+    ADD CONSTRAINT metrics_reputation_pk
+        PRIMARY KEY (day, cause);
+
+-- prepopulate the table with data from the not current day. After this the metric service takes over.
+INSERT INTO repbot_schema.metrics_reputation
+SELECT received::DATE AS day,
+       cause,
+       COUNT(1)       AS count
+FROM repbot_schema.reputation_log
+WHERE received::DATE != NOW()::DATE
+GROUP BY day, cause
+ORDER BY day DESC;
+
+-- We now need to recreate all views which were based on the reputation log itself
+CREATE OR REPLACE VIEW repbot_schema.metrics_reputation_type_week AS
+SELECT DATE_TRUNC('week', day)::DATE AS week,
+       cause,
+       SUM(count)::BIGINT            AS count
+FROM repbot_schema.metrics_reputation
+GROUP BY week, cause
+ORDER BY week DESC, cause;
+
+CREATE OR REPLACE VIEW repbot_schema.metrics_reputation_type_month AS
+SELECT DATE_TRUNC('month', day)::DATE AS month,
+       cause,
+       SUM(count)::BIGINT             AS count
+FROM repbot_schema.metrics_reputation
+GROUP BY month, cause
+ORDER BY month DESC;
+
+CREATE OR REPLACE VIEW repbot_schema.metrics_reputation_dow_week AS
+SELECT DATE_TRUNC('week', day)::DATE AS week,
+       EXTRACT(ISODOW FROM day)      AS dow,
+       SUM(count)::BIGINT            AS count
+FROM repbot_schema.metrics_reputation
+GROUP BY week, dow
+ORDER BY week DESC, dow ASC;
+
+CREATE OR REPLACE VIEW repbot_schema.metrics_reputation_week AS
+SELECT DATE_TRUNC('week', day)::DATE AS week,
+       SUM(count)::BIGINT            AS count
+FROM repbot_schema.metrics_reputation
+GROUP BY week
+ORDER BY week DESC;
+
+CREATE OR REPLACE VIEW repbot_schema.metrics_reputation_month AS
+SELECT DATE_TRUNC('month', day)::DATE AS month,
+       SUM(count)::BIGINT             AS count
+FROM repbot_schema.metrics_reputation
+GROUP BY month
+ORDER BY month DESC;
+
+-- Create a table showing the total of saved reputation in the database
+CREATE TABLE repbot_schema.metrics_reputation_count
+(
+    day     DATE              NOT NULL
+        CONSTRAINT metrics_reputation_count_pk
+            PRIMARY KEY,
+    count   INTEGER DEFAULT 0 NOT NULL,
+    added   INTEGER DEFAULT 0 NOT NULL,
+    removed INTEGER DEFAULT 0 NOT NULL
+);
+
+-- Prepopulate counts
+INSERT INTO repbot_schema.metrics_reputation_count
```

Comment:
Scope of this issue is to not perform a count since a count does not represent deletions. If we delete from the table all graphs will simple get a bit less but you wont see any signs of deletion since the view is not aware of those changes.

If a guild gets deleted and the data from the reputation log gets deleted, there will be no visible change to the total development. Aggregation is good to get a view of the current state but actually horrible when you want some kind of timeline.